pilot/cmd/pilot-agent/main.go

```diff
@@ -92,6 +99,9 @@ var (
 	podNamespaceVar      = env.RegisterStringVar("POD_NAMESPACE", "", "")
 	istioNamespaceVar    = env.RegisterStringVar("ISTIO_NAMESPACE", "", "")
 	kubeAppProberNameVar = env.RegisterStringVar(status.KubeAppProberEnvName, "", "")
+	sdsEnabledVar        = env.RegisterBoolVar("SDS_ENABLED", false, "")
```

Comment:
I don't like bool variables, but for now it'll do. 
However - the goal of this env variable is to also allow user fine control over sds deployment, by using pod annotations ( which get auto-translated to env ). 

But we can start with this - just please fill in the third param, should be a doc. And mention this is internal and temporary.